### DISCUSSION TITLE
Upload a rule to docker image using an API

### DISCUSSION THREAD

**Discussion by ishaan812** [gorules/zen#100] [Q&A] [answered]
Hey, I wanted to integrate with self-hosted go-rules. Although, wanted to only access the docker container as a REST service. We also could'nt use one of the ways provided since we use Golang everywhere. The evaluation API would work but loading the rule onto it using the UI would be an issue. How should I go about this. Should we have a seperate react-based interface to load the rules or is there already support to do this using REST already.

**Comment by stefan-gorules** [comment, marked as answer]
Hi Ishaan,

Would you be able to clarify? We don't have native GoLang support, however you use Golangs to make [HTTP request](https://blog.logrocket.com/making-http-requests-in-go/) to self-hosted API version.

Here's the details on how to use REST API: https://gorules.io/docs/cloud/evaluation-api.

Note that this is for self-hosted version which exposes REST services and provides you with UI for rules management. Zen Engine itself as a library doesn't work currently with GoLang.

**Reply by ishaan812** [reply]
Hey Stefan,
I am aware of the evaluation api. Although the specific usecase I was looking for is uploading a decision to gorules\ changing status of it from drafting to published maybe using an API. Any way I can do that yet?

**Reply by stefan-gorules** [reply]
Hi,

Thanks for clarifying. This will be available with PATs (personal access token). You will be able to call APIs for uploading decisions/rules, updating their statuses, generating and publishing releases etc. We're planning to have it in Q1 2024.
